compute_tools/src/extension_server.rs

```diff
@@ -0,0 +1,244 @@
+    // pg_config --version returns a (platform specific) human readable string
+    // such as "PostgreSQL 15.4". We parse this to v14/v15
+    let human_version = get_pg_config("--version", pgbin);
+    if human_version.contains("15") {
+        return "v15".to_string();
+    } else if human_version.contains("14") {
+        return "v14".to_string();
+    }
+    panic!("Unsuported postgres version {human_version}");
+}
+
+// download control files for enabled_extensions
+// return the paths in s3 to the archives containing the actual extension files
+// for use in creating the extension
+pub async fn get_available_extensions(
+    remote_storage: &GenericRemoteStorage,
+    pgbin: &str,
+    pg_version: &str,
+    enabled_extensions: &[String],
+    build_tag: &str,
+) -> Result<HashMap<String, RemotePath>> {
+    let local_sharedir = Path::new(&get_pg_config("--sharedir", pgbin)).join("extension");
+    let index_path = format!("{build_tag}/{pg_version}/ext_index.json");
+    let index_path = RemotePath::new(Path::new(&index_path)).context("error forming path")?;
+    info!("download ext_index.json from: {:?}", &index_path);
+
+    let mut download = remote_storage.download(&index_path).await?;
+    let mut ext_idx_buffer = Vec::new();
+    download
+        .download_stream
+        .read_to_end(&mut ext_idx_buffer)
+        .await?;
+    let ext_index_str = str::from_utf8(&ext_idx_buffer).expect("error parsing json");
+    let ext_index_full: Value = serde_json::from_str(ext_index_str)?;
+    let ext_index_full = ext_index_full.as_object().context("error parsing json")?;
+    info!("ext_index: {:?}", &ext_index_full);
+
+    info!("enabled_extensions: {:?}", enabled_extensions);
+    let mut ext_remote_paths = HashMap::new();
+    let mut file_create_tasks = Vec::new();
+    for extension in enabled_extensions {
+        let ext_data = ext_index_full[extension]
+            .as_object()
+            .context("error parsing json")?;
+
+        let control_files = ext_data["control_data"]
+            .as_object()
+            .context("error parsing json")?;
+        for (control_file, control_contents) in control_files {
+            let control_path = local_sharedir.join(control_file);
+            let control_contents_str = control_contents.as_str().context("error parsing json")?;
+            info!("writing file {:?}{:?}", control_path, control_contents);
+            file_create_tasks.push(tokio::fs::write(control_path, control_contents_str));
+        }
+
+        let ext_archive_path = ext_data["archive_path"]
+            .as_str()
+            .context("error parsing json")?;
+        ext_remote_paths.insert(
+            extension.to_string(),
+            RemotePath::from_string(ext_archive_path)?,
+        );
+    }
+    let results = join_all(file_create_tasks).await;
+    for result in results {
+        result?;
+    }
+    Ok(ext_remote_paths)
+}
+
+// download all sqlfiles (and possibly data files) for a given extension name
```

Comment:
Please, update comment to reflect that we have an archive now.

Reply:
done